cmd/exo/cmd/sos_upload.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	minio "github.com/minio/minio-go"
+	"github.com/spf13/cobra"
+)
+
+// uploadCmd represents the upload command
+var sosUploadCmd = &cobra.Command{
+	Use:     "upload <bucket name> <local file path> [remote file path]",
+	Short:   "Upload an object into a bucket",
+	Aliases: gUploadAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return cmd.Usage()
+		}
+
+		var remoteFilePath string
+		if len(args) > 2 {
+			remoteFilePath = args[2]
+		}
+
+		minioClient, err := newMinioClient(sosZone)
+		if err != nil {
+			return err
+		}
+
+		location, err := minioClient.GetBucketLocation(args[0])
+		if err != nil {
+			return err
+		}
+
+		minioClient, err = newMinioClient(location)
+		if err != nil {
+			return err
+		}
+
+		// Upload the  file
+		bucketName := args[0]
+		objectName := filepath.Base(args[1])
+		filePath := args[1]
+
+		if strings.HasSuffix(remoteFilePath, "/") {
+			remoteFilePath = remoteFilePath + objectName
+		}
+
+		file, err := os.Open(filePath)
+		if err != nil {
+			return err
+		}
+		defer file.Close() // nolint: errcheck
+
+		// Only the first 512 bytes are used to sniff the content type.
+		buffer := make([]byte, 512)
+		_, err = file.Read(buffer)
+		if err != nil {
+			return err
+		}
+
+		if remoteFilePath == "" {
+			remoteFilePath = filePath
+		}
+
+		contentType := http.DetectContentType(buffer)
+
+		// Upload object with FPutObject
+		n, err := minioClient.FPutObjectWithContext(gContext, bucketName, remoteFilePath, filePath, minio.PutObjectOptions{ContentType: contentType})
```

Comment:
https://github.com/minio/minio-go/blob/2508a98a820e993b0307bbac4db9731e4e6f89a0/api-put-object.go#L37

By bored to death, I meant that it should show a progress bar or something along those lines.

Reply:
okay thanks